BGP CP: API Helper Functions Cleanup

danehans opened the pull request:
- Moves functions next to types.
- Renames functions to ParseXXX
- Adds godocs.

rastislavs commented on pkg/bgpv1/api/conversions.go:
```diff
@@ -16,85 +16,76 @@ import (
-		return types.AfiLS
-	case "opaque":
-		return types.AfiOpaque
+func ToAPIAfi(afi types.Afi) string {
```
Why this adds back the code removed in https://github.com/cilium/cilium/pull/28035?

danehans replied:
Thanks for catching this. It must have slipped back in during a rebase.

danehans committed:
BGP CP: API Helper Functions Cleanup

- Moves functions next to types.
- Renames functions to ParseXXX
- Adds godocs.

Signed-off-by: Daneyon Hansen <daneyon.hansen@solo.io>

rastislavs reviewed (approved):
LGTM
